doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,246 @@
+
+## Summary
+
+This enhancement proposal outlines the SDK scorecard’s design for running of custom tests, specifically tests that are outside of what Red Hat includes by default within the scorecard’s tests.   The proposed design would provide both a means to write custom tests and also convert the existing scorecard tests (basic and olm) into the custom test format.
+
+## Motivation
+
+The scorecard needs a robust means of enabling end-users of the SDK to write and execute their own operator tests.   Ideally, the custom test framework would decouple test logic from within the SDK itself, making the SDK and scorecard logic in particular more flexible and maintainable in the long run.
+
+## Goals
+
+ * Document a design for scorecard to execute end-user or custom tests.
+
+ * Document stories/steps required to deliver scorecard custom test functionality.
+
+### Non-Goals
+
+ * This is a design proposal and does not include a proof-of-concept.
+
+## Proposal
+
+This design proposal is to establish a pattern for implementing custom scorecard tests and also support the goal of migrating existing scorecard tests into the same format as the custom tests.
+
+With this proposal, the scorecard will change as follows:
+
+ * scorecard will not start the operator being tested
+ * scorecard will not create Custom Resources on behalf of test
+ * scorecard will execute tests as Pods based on scorecard configuration
+ * scorecard will require a valid bundle image or on-disk bundle to determine what information will be passed to scorecard tests
+ * scorecard will determine test configurations by a YAML/DSL file that is either passed on the command line or included in the bundle contents
+ * scorecard will only run OLM tests if OLM manifests are found in the bundle
+ * scorecard command syntax will look as follows:
+   * operator-sdk scorecard --dsl <point-to-yaml> --bundle <bundle-path/image> -l <selector>
+
+### User Stories
+
+#### Story 1
+
+I am as an Operator developer, I'd like to be able to construct custom tests and execute them via the scorecard.
+
+#### Story 2
+
+Scorecard will include an example custom test that end users could use as a reference when writing their custom tests.
+
+#### Story 3
+
+Scorecard internal tests will be converted to the custom test format and therefore be externalized from the scorecard binary itself.
+
+#### Story 4
+
+Scorecard would process tests based on an on-disk directory or bundle image that contains the scorecard configuration.
+
+#### Story 5 
+
+- I am as an Operator developer, would like to be able to create tests using the same Assert syntax adopted by [Kuttl](https://github.com/kudobuilder/kuttl) and check its results aggregated within scorecard results. 
+
+**Implementation Notes**
+
+The Scorecard would support the creation of a Kuttl test image and the execute tests via Kuttl, however, it would be transparent for users since its results will be integrated into scorecard results.
```

Comment:
nit: and then execute

Reply:
the commit just now fixes that type-o, thanks.